ExploreASL_Initialize.m

```diff
@@ -27,287 +21,249 @@
         else
-            pathstr = input('Provide foldername where ExploreASL is installed, including ": ');
+            x.dataParType = 'sourceStructure';
         end
-        if sum(pathstr==0) || ~exist(fullfile(pathstr,'ExploreASL_Master.m'),'file')
-            return
-        end
-        x.MyPath = pathstr;
+    else
+        x.dataParType = 'unknown';
     end
-else
-    % In deployed mode set the ExploreASL directory in the ctf archive
-    [files,~] = spm_select('FPListRec',ctfroot,'ExploreASL_Master*'); % Find the path of the master files within the ctf archive
-    curPathCTF = fileparts(files(1,:)); % Get the path
-    x.MyPath = fullfile(curPathCTF); % curPathCTF = ExploreASL path
     
-    BreakString = '==============================================================================================\n';
-    fprintf(BreakString);
-    fprintf('ctfroot:  %s\n', ctfroot);
-    fprintf('x.MyPath: %s\n', x.MyPath);
-    fprintf(BreakString);
-
-end
-
-% Go to ExploreASL folder
-cd(x.MyPath);
-
-
-%% Add ExploreASL paths
-if ~isdeployed
-% Define paths (should be equal when loading data or only initializing
-% ExploreASL
-    addpath(x.MyPath);
-
-	%fullfile('Development','dicomtools'), ...
-    subfolders_to_add = {'Functions', 'mex', 'Modules', 'Testing',...
-                            fullfile('Modules', 'SubModule_Structural'), ...
-                            fullfile('Modules', 'SubModule_ASL'), ...
-                            fullfile('Modules', 'SubModule_Population'), ...
-                            'Development', 'External', ...
-                            fullfile('External','isnear'), ...
-                            fullfile('External','DCMTK'), ...
-                            fullfile('Testing', 'UnitTests'), ...
-                            fullfile('External','ExploreQC'), ...
-							fullfile('External','SPMmodified'), ...
-                            fullfile('External','SPMmodified','matlabbatch'),...                            
-                            fullfile('External','SPMmodified','xASL'),...
-                            fullfile('External','SPMmodified','toolbox','cat12'), ...
-							fullfile('External','SPMmodified','toolbox','LST'), ...
-							fullfile('External','SPMmodified','toolbox','OldNorm') ...
-							genpath(fullfile('External','bids-matlab'))};
-
-    for ii=1:length(subfolders_to_add)
-        addpath(fullfile(x.MyPath,subfolders_to_add{ii}));
+    % Check if pipeline should be run, but there is no DataParFile
+    if x.ProcessData && (~isfield(x,'DataParPath') || ~exist(x.DataParPath,'file'))
+        x.DataParPath = input('Please insert the path to your DataParFile: ');
     end
-end
 
+    % Recheck the DataParFile, which is possibly not a file or does not exist
+    if ~exist(x.DataParPath,'file') && ~exist(x.DataParPath,'dir') % file = without BIDS Import, dir = BIDS import 
+        fprintf('DataPar file does not exist, ExploreASL will only be initialized...\n');
```

Comment:
This will not be printed if you don't specify `DataParPath` at all right?
It should still be possible to initialize ExploreASL without getting this output, this output should only there with an invalid `DataParPath` indeed.

Reply:
Oh... here's a mistake in the comment anyway. I'll quickly check it later this afternoon.

Reply:
Should be fixed in 5d88126. I also removed the quite verbose user feedback on default. It's really helpful during debugging, because there are so many different use cases.

Reply:
Damn... I know what you want to do, but it's kind of hard to code. So with the changes in c23eac5, there is no warning that the file does not exist, if you only initialize using `ExploreASL_Master();` or `ExploreASL_Initialize();`, since I check if either the import or the processing is turned on. 
If we turn on processing, then we get the CLI request and if this file doesn't exist either, the warning appears:
```
>> [x] = ExploreASL_Master('I dont exist', 0, 1, 1);
Please insert the path to your DataParFile: 'I dont exist either'
DataPar file does not exist, ExploreASL will only be initialized...
```
If the inserted file exists, the processing will start as usual:
```
>> [x] = ExploreASL_Master('I dont exist', 0, 1, 1);
Please insert the path to your DataParFile: '.\analysis\DataParameters_LowQ.json'
```
This doesn't work for the import workflow right now, meaning we don't ask a second time there:
```
>> [x] = ExploreASL_Master('I dont exist', 1, 0, 1);
DataPar file does not exist, ExploreASL will only be initialized...
```
And if processing and import are turned off, but the file doesn't exist, there is no warning (fallback solution for data loading?):
```
[x] = ExploreASL_Master('I dont exist', 0, 0, 1);
```
But there's the question if we need it there...

Reply:
I like this example a lot, but do I now understand correctly that the first argument of `ExploreASL_Master` (i.e. the dataParFile) can be used as original dataPar file (for processing) as well as import-source file (for importing)?
That is perhaps asking for trouble... Unless we always use the "import-source file" for importing and then convert it later to an "original dataPar" file. 

(I know Jan uses a dataPar file for doing the import, and that this can be fine, but we are moving towards having the data parameters in JSON sidecars of the NIfTIs, and the study processing settings in the original dataPar (to make it confusing ;) so perhaps we should at least change the name of the original dataPar to processingPar or something :)

Reply:
That's actually pretty straight forward. We open the json. If the "root" variable is called "x", then it's a DataParFile.

Reply:
Just a bit of additional explanation:
(1) If you run the ImportModules, ExploreASL expects the `sourceStructure.json` file.
(2) If all ImportModules are set to false, ExploreASL expects a `DataPar.json` file.
It's the best way, at least in my opinion. Originally, I gave the Import Modules only the directory and it searched for possible files, but that seems a bit more risky. Jan then told me to pass on the `sourceStructure.json` file and I think it works pretty well.